runtimes/peregrine/src/asset_swap/mod.rs

```diff
@@ -0,0 +1,107 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2024 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use frame_support::ensure;
+use pallet_asset_swap::traits::SwapHooks;
+use runtime_common::{AccountId, Balance};
+use xcm::{
+	v3::{Junction, MultiLocation},
+	VersionedMultiLocation,
+};
+
+use crate::Runtime;
+
+const LOG_TARGET: &str = "runtime::peregrine::asset-swap::RestrictTransfersToSameUser";
+
+/// Check requiring the beneficiary be a single `AccountId32` junction
+/// containing the same account ID as the account on this chain initiating the
+/// swap.
+pub struct RestrictSwapDestinationToSelf;
```

Comment:
Do you think, this `SwapHook` implementation could be interesting for other teams?

Reply:
We'll find out 😊 I wouldn't commit to have it into other crates for now, to be honest. We can always export it there if we see many projects will want to enforce the same requirement.